some minor clippy fixes

matthiaskrgr opened the pull request.

calebcartwright reviewed (approved):
Thanks! This is the smallest one of these in a while, I take that as a good sign and meaning we're gradually reducing all the issues :smile:

matthiaskrgr commented:
:sweat_smile: 
These were mostly new ones popping up.
The ones that are currently remaining are mostly questionable imho

````
warning: using `write!()` with a format string that ends in a single newline
   --> src/config/file_lines.rs:166:21
    |
166 |                     write!(f, "{}\n", ranges.iter().format(", "))?;
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `#[warn(clippy::write_with_newline)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#write_with_newline
help: use `writeln!()` instead
    |
166 -                     write!(f, "{}\n", ranges.iter().format(", "))?;
166 +                     writeln!(f, "{}", ranges.iter().format(", "))?;
    |

warning: this `else { if .. }` block can be collapsed
  --> src/expr.rs:81:20
   |
81 |               } else {
   |  ____________________^
82 | |                 if let LitKind::StrRaw(_) = l.token.kind {
83 | |                     Some(context.snippet(l.span).trim().into())
84 | |                 } else {
85 | |                     None
86 | |                 }
87 | |             }
   | |_____________^
   |
   = note: `#[warn(clippy::collapsible_else_if)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#collapsible_else_if
help: collapse nested if block
   |
81 ~             } else if let LitKind::StrRaw(_) = l.token.kind {
82 +                 Some(context.snippet(l.span).trim().into())
83 +             } else {
84 +                 None
85 +             }
   |

warning: this `else { if .. }` block can be collapsed
   --> src/matches.rs:430:24
    |
430 |                   } else {
    |  ________________________^
431 | |                     if semicolon_for_expr(context, body) {
432 | |                         ";"
433 | |                     } else {
434 | |                         ""
435 | |                     }
436 | |                 };
    | |_________________^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#collapsible_else_if
help: collapse nested if block
    |
430 ~                 } else if semicolon_for_expr(context, body) {
431 +                     ";"
432 +                 } else {
433 +                     ""
434 ~                 };
    |

warning: this `else { if .. }` block can be collapsed
   --> src/patterns.rs:331:16
    |
331 |           } else {
    |  ________________^
332 | |             if !fields_str.is_empty() {
333 | |                 // there are preceding struct fields being matched on
334 | |                 if has_trailing_comma {
...   |
339 | |             }
340 | |         }
    | |_________^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#collapsible_else_if
help: collapse nested if block
    |
331 ~         } else if !fields_str.is_empty() {
332 +             // there are preceding struct fields being matched on
333 +             if has_trailing_comma {
334 +                 fields_str.push(' ');
335 +             } else {
336 +                 fields_str.push_str(", ");
  ...

warning: this `if` statement can be collapsed
   --> src/types.rs:660:17
    |
660 | /                 if context.inside_macro() && bounds.len() == 1 {
661 | |                     if context.snippet(self.span).ends_with('+') && !res.ends_with('+') {
662 | |                         res.push('+');
663 | |                     }
664 | |                 }
    | |_________________^
    |
    = note: `#[warn(clippy::collapsible_if)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#collapsible_if
help: collapse nested if block
    |
660 ~                 if context.inside_macro() && bounds.len() == 1 && context.snippet(self.span).ends_with('+') && !res.ends_with('+') {
661 +                     res.push('+');
662 +                 }
    |

warning: manual implementation of `split_once`
   --> src/utils.rs:215:23
    |
215 |     unicode_str_width(s.splitn(2, '\n').next().unwrap_or(""))
    |                       ^^^^^^^^^^^^^^^^^^^^^^^^ help: try this: `Some(s.split_once('\n').map_or(s, |x| x.0))`
    |
    = note: `#[warn(clippy::manual_split_once)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#manual_split_once

warning: unnecessary use of `splitn`
   --> src/utils.rs:215:23
    |
215 |     unicode_str_width(s.splitn(2, '\n').next().unwrap_or(""))
    |                       ^^^^^^^^^^^^^^^^^ help: try this: `s.split('\n')`
    |
    = note: `#[warn(clippy::needless_splitn)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_splitn

warning: manual implementation of `rsplit_once`
   --> src/utils.rs:221:23
    |
221 |     unicode_str_width(s.rsplitn(2, '\n').next().unwrap_or(""))
    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^ help: try this: `s.rsplit_once('\n').map(|x| x.1)`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#manual_split_once

warning: unnecessary use of `rsplitn`
   --> src/utils.rs:221:23
    |
221 |     unicode_str_width(s.rsplitn(2, '\n').next().unwrap_or(""))
    |                       ^^^^^^^^^^^^^^^^^^ help: try this: `s.rsplit('\n')`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_splitn

warning: unnecessary closure used to substitute value for `Option::None`
   --> src/attr.rs:274:15
    |
274 |     let len = comment_begin.unwrap_or_else(|| comment.len());
    |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use `unwrap_or` instead: `comment_begin.unwrap_or(comment.len())`
    |
    = note: `#[warn(clippy::unnecessary_lazy_evaluations)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_lazy_evaluations

warning: this function has too many arguments (8/7)
  --> src/closures.rs:26:1
   |
26 | / pub(crate) fn rewrite_closure(
27 | |     capture: ast::CaptureBy,
28 | |     is_async: &ast::Async,
29 | |     movability: ast::Movability,
...  |
34 | |     shape: Shape,
35 | | ) -> Option<String> {
   | |___________________^
   |
   = note: `#[warn(clippy::too_many_arguments)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#too_many_arguments

warning: this function has too many arguments (8/7)
   --> src/closures.rs:229:1
    |
229 | / fn rewrite_closure_fn_decl(
230 | |     capture: ast::CaptureBy,
231 | |     asyncness: &ast::Async,
232 | |     movability: ast::Movability,
...   |
237 | |     shape: Shape,
238 | | ) -> Option<(String, usize)> {
    | |____________________________^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#too_many_arguments

warning: methods with the following characteristics: (`to_*` and `self` type is `Copy`) usually take `self` by value
   --> src/comment.rs:123:33
    |
123 |     pub(crate) fn to_str_tuplet(&self) -> (&'a str, &'a str, &'a str) {
    |                                 ^^^^^
    |
    = note: `#[warn(clippy::wrong_self_convention)]` on by default
    = help: consider choosing a less ambiguous name
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#wrong_self_convention

warning: all variants have the same postfix: `Line`
  --> src/config/options.rs:39:1
   |
39 | #[config_type]
   | ^^^^^^^^^^^^^^
   |
   = note: `#[warn(clippy::enum_variant_names)]` on by default
   = help: remove the postfixes and use full paths to the variants instead of glob imports
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#enum_variant_names
   = note: this warning originates in the attribute macro `config_type` (in Nightly builds, run with -Z macro-backtrace for more info)

warning: this creates an owned instance just for comparison
   --> src/config/mod.rs:154:31
    |
154 |     required_version: String, env!("CARGO_PKG_VERSION").to_owned(), false,
    |                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `*env!("CARGO_PKG_VERSION")`
    |
    = note: `#[warn(clippy::cmp_owned)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#cmp_owned

warning: this function has too many arguments (8/7)
    --> src/expr.rs:1523:1
     |
1523 | / fn rewrite_struct_lit<'a>(
1524 | |     context: &RewriteContext<'_>,
1525 | |     path: &ast::Path,
1526 | |     qself: Option<&ast::QSelf>,
...    |
1531 | |     shape: Shape,
1532 | | ) -> Option<String> {
     | |___________________^
     |
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#too_many_arguments

warning: this function has too many arguments (8/7)
    --> src/expr.rs:1982:1
     |
1982 | / pub(crate) fn rewrite_assign_rhs_with_comments<S: Into<String>, R: Rewrite>(
1983 | |     context: &RewriteContext<'_>,
1984 | |     lhs: S,
1985 | |     ex: &R,
...    |
1990 | |     allow_extend: bool,
1991 | | ) -> Option<String> {
     | |___________________^
     |
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#too_many_arguments

warning: all variants have the same postfix: `Item`
    --> src/items.rs:1495:1
     |
1495 | / pub(crate) enum ItemVisitorKind<'a> {
1496 | |     Item(&'a ast::Item),
1497 | |     AssocTraitItem(&'a ast::AssocItem),
1498 | |     AssocImplItem(&'a ast::AssocItem),
1499 | |     ForeignItem(&'a ast::ForeignItem),
1500 | | }
     | |_^
     |
     = help: remove the postfixes and use full paths to the variants instead of glob imports
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#enum_variant_names

warning: this function has too many arguments (8/7)
    --> src/items.rs:2515:1
     |
2515 | / fn rewrite_params(
2516 | |     context: &RewriteContext<'_>,
2517 | |     params: &[ast::Param],
2518 | |     one_line_budget: usize,
...    |
2523 | |     variadic: bool,
2524 | | ) -> Option<String> {
     | |___________________^
     |
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#too_many_arguments

warning: this function has too many arguments (9/7)
    --> src/items.rs:2838:1
     |
2838 | / fn rewrite_where_clause(
2839 | |     context: &RewriteContext<'_>,
2840 | |     where_clause: &ast::WhereClause,
2841 | |     brace_style: BraceStyle,
...    |
2847 | |     where_clause_option: WhereClauseOption,
2848 | | ) -> Option<String> {
     | |___________________^
     |
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#too_many_arguments

warning: unnecessary closure used to substitute value for `Option::None`
   --> src/lists.rs:651:16
    |
651 |           return post_snippet
    |  ________________^
652 | |             .find_uncommented(terminator)
653 | |             .unwrap_or_else(|| post_snippet.len());
    | |__________________________________________________^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_lazy_evaluations
help: use `unwrap_or` instead
    |
651 ~         return post_snippet
652 ~             .find_uncommented(terminator).unwrap_or(post_snippet.len());
    |

warning: unnecessary closure used to substitute value for `Option::None`
   --> src/lists.rs:709:25
    |
709 |       let first_newline = test_snippet
    |  _________________________^
710 | |         .find('\n')
711 | |         .unwrap_or_else(|| test_snippet.len());
    | |______________________________________________^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_lazy_evaluations
help: use `unwrap_or` instead
    |
709 ~     let first_newline = test_snippet
710 ~         .find('\n').unwrap_or(test_snippet.len());
    |

warning: unnecessary closure used to substitute value for `Option::None`
   --> src/lists.rs:714:17
    |
714 |       let first = test_snippet
    |  _________________^
715 | |         .find(|c: char| !c.is_whitespace())
716 | |         .unwrap_or_else(|| test_snippet.len());
    | |______________________________________________^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_lazy_evaluations
help: use `unwrap_or` instead
    |
714 ~     let first = test_snippet
715 ~         .find(|c: char| !c.is_whitespace()).unwrap_or(test_snippet.len());
    |

warning: use of `unwrap_or` followed by a function call
   --> src/modules.rs:129:49
    |
129 |             FileName::Real(ref p) => p.parent().unwrap_or(Path::new("")).to_path_buf(),
    |                                                 ^^^^^^^^^^^^^^^^^^^^^^^^ help: try this: `unwrap_or_else(|| Path::new(""))`
    |
    = note: `#[warn(clippy::or_fun_call)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#or_fun_call

warning: methods called `into_*` usually take `self` by value
   --> src/overflow.rs:197:31
    |
197 |     fn into_overflowable_item(&'a self) -> OverflowableItem<'a>;
    |                               ^^^^^^^^
    |
    = help: consider choosing a less ambiguous name
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#wrong_self_convention

warning: this function has too many arguments (10/7)
   --> src/overflow.rs:340:5
    |
340 | /     fn new<T: 'a + IntoOverflowableItem<'a>>(
341 | |         context: &'a RewriteContext<'_>,
342 | |         items: impl Iterator<Item = &'a T>,
343 | |         ident: &'a str,
...   |
350 | |         custom_delims: Option<(&'a str, &'a str)>,
351 | |     ) -> Context<'a> {
    | |____________________^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#too_many_arguments

warning: manual implementation of `split_once`
   --> src/overflow.rs:436:43
    |
436 |             let rewrite_first_line = Some(rewrite.splitn(2, '\n').next().unwrap().to_owned());
    |                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try this: `rewrite.split_once('\n').map_or(&*rewrite, |x| x.0)`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#manual_split_once

warning: unnecessary use of `splitn`
   --> src/overflow.rs:436:43
    |
436 |             let rewrite_first_line = Some(rewrite.splitn(2, '\n').next().unwrap().to_owned());
    |                                           ^^^^^^^^^^^^^^^^^^^^^^^ help: try this: `rewrite.split('\n')`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_splitn

warning: very complex type used. Consider factoring parts into `type` definitions
  --> src/parse/macros/lazy_static.rs:12:6
   |
12 | ) -> Option<Vec<(ast::Visibility, symbol::Ident, P<ast::Ty>, P<ast::Expr>)>> {
   |      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(clippy::type_complexity)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#type_complexity

warning: explicit lifetimes given in parameter types where they could be elided (or replaced with `'_` if needed by type declaration)
  --> src/parse/macros/mod.rs:17:1
   |
17 | fn build_stream_parser<'a>(sess: &'a ParseSess, tokens: TokenStream) -> Parser<'a> {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(clippy::needless_lifetimes)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_lifetimes

warning: redundant closure
  --> src/parse/macros/mod.rs:50:9
   |
50 |         |x: ptr::P<ast::Expr>| Some(x)
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: replace the closure with the function itself: `Some`
   |
   = note: `#[warn(clippy::redundant_closure)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#redundant_closure

warning: redundant closure
  --> src/parse/macros/mod.rs:55:9
   |
55 |         |x: ptr::P<ast::Ty>| Some(x)
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: replace the closure with the function itself: `Some`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#redundant_closure

warning: redundant closure
  --> src/parse/macros/mod.rs:60:9
   |
60 |         |x: ptr::P<ast::Pat>| Some(x)
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: replace the closure with the function itself: `Some`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#redundant_closure

warning: very complex type used. Consider factoring parts into `type` definitions
   --> src/parse/parser.rs:112:10
    |
112 |     ) -> Result<(Vec<ast::Attribute>, Vec<ptr::P<ast::Item>>, Span), ParserError> {
    |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#type_complexity

warning: methods with the following characteristics: (`to_*` and `self` type is `Copy`) usually take `self` by value
  --> src/shape.rs:68:29
   |
68 |     pub(crate) fn to_string(&self, config: &Config) -> Cow<'static, str> {
   |                             ^^^^^
   |
   = help: consider choosing a less ambiguous name
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#wrong_self_convention

warning: methods with the following characteristics: (`to_*` and `self` type is `Copy`) usually take `self` by value
  --> src/shape.rs:72:42
   |
72 |     pub(crate) fn to_string_with_newline(&self, config: &Config) -> Cow<'static, str> {
   |                                          ^^^^^
   |
   = help: consider choosing a less ambiguous name
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#wrong_self_convention

warning: methods with the following characteristics: (`to_*` and `self` type is `Copy`) usually take `self` by value
  --> src/shape.rs:76:24
   |
76 |     fn to_string_inner(&self, config: &Config, offset: usize) -> Cow<'static, str> {
   |                        ^^^^^
   |
   = help: consider choosing a less ambiguous name
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#wrong_self_convention

warning: methods with the following characteristics: (`to_*` and `self` type is `Copy`) usually take `self` by value
   --> src/shape.rs:275:42
    |
275 |     pub(crate) fn to_string_with_newline(&self, config: &Config) -> Cow<'static, str> {
    |                                          ^^^^^
    |
    = help: consider choosing a less ambiguous name
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#wrong_self_convention

warning: manual implementation of `split_once`
   --> src/visitor.rs:276:31
    |
276 |             let first_lines = comment_snippet.splitn(2, '/').next().unwrap_or("");
    |                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try this: `Some(comment_snippet.split_once('/').map_or(comment_snippet, |x| x.0))`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#manual_split_once

warning: unnecessary use of `splitn`
   --> src/visitor.rs:276:31
    |
276 |             let first_lines = comment_snippet.splitn(2, '/').next().unwrap_or("");
    |                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try this: `comment_snippet.split('/')`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_splitn

warning: unnecessary closure used to substitute value for `Option::None`
   --> src/lib.rs:392:21
    |
392 |       let block_len = formatted
    |  _____________________^
393 | |         .snippet
394 | |         .rfind('}')
395 | |         .unwrap_or_else(|| formatted.snippet.len());
    | |___________________________________________________^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_lazy_evaluations
help: use `unwrap_or` instead
    |
392 ~     let block_len = formatted
393 +         .snippet
394 ~         .rfind('}').unwrap_or(formatted.snippet.len());
    |

warning: `rustfmt-nightly` (lib) generated 41 warnings
warning: field assignment outside of initializer for an instance created with Default::default()
   --> src/bin/main.rs:524:9
    |
524 |         options.verbose = matches.opt_present("verbose");
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `#[warn(clippy::field_reassign_with_default)]` on by default
note: consider initializing the variable with `GetOptsOptions { verbose: matches.opt_present("verbose"), quiet: matches.opt_present("quiet"), ..Default::default() }` and removing relevant reassignments
   --> src/bin/main.rs:523:9
    |
523 |         let mut options = GetOptsOptions::default();
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#field_reassign_with_default

warning: `rustfmt-nightly` (bin "rustfmt") generated 1 warning
    Finished dev [unoptimized + debuginfo] target(s) in 11.68s
````
